javascript/features/death_match/death_match_manager.js

```diff
@@ -102,11 +94,28 @@ export class DeathMatchManger {
             player.respawn();
         }
 
-        this.resetTeamScoreIfZoneEmpty(zone);
-
         this.showStats(player);
     }
 
+    restoreDefaultPlayerStatus(player) {
+        if (!this.playersInDeathMatch_.has(player))
+            return;
+        
+        const zone = this.playersInDeathMatch_.get(player);
+        const location = DeathMatchLocation.getById(zone);
+
+        this.playersInDeathMatch_.delete(player);
+        this.playerTeam_.delete(player);
+        
+        player.activity = Player.PLAYER_ACTIVITY_NONE;
+        player.team = Player.kNoTeam;
+        if(!isNaN(location.gravity)) 
+            player.gravity = Player.kDefaultGravity;
+
```

Comment:
Given #745, should lag shot be in here too?

Reply:
I don't think we want to do it here as this is also passed by when switching from DM zone.

Resulting in it switching to 2 and 0 in the same frame. (What could potentially cause the wrong package to arrive later due to desync I assume?)

I added it to the leave with abuse though. That's where we're missing it.